e2e/head_test.go

```diff
@@ -0,0 +1,407 @@
+package e2e
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"gotest.tools/v3/icmd"
+)
+
+//head (without anything -> error)
+
+func TestHead(t *testing.T) {
+	t.Parallel()
+
+	_, s5cmd := setup(t)
+
+	cmd := s5cmd("head")
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+}
+
+// ------------------ Bucket Tests ------------------
+
+// head bucket
+
+func TestHeadBucket(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	cmd := s5cmd("head", fmt.Sprintf("s3://%v", bucket))
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 0, Out: bucket})
+}
+
+// head bucket (non-existent)
+
+func TestHeadBucketNonExistent(t *testing.T) {
+	t.Parallel()
+
+	_, s5cmd := setup(t)
+
+	cmd := s5cmd("head", "s3://non-existent-bucket")
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+}
+
+// --json head bucket
+
+func TestHeadBucketJSON(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	cmd := s5cmd("--json", "head", fmt.Sprintf("s3://%v", bucket))
+	result := icmd.RunCmd(cmd)
+
+	//{"created_at":"2021/09/01 00:00:00","name":"bucket"}
+
+	result.Assert(t, icmd.Expected{ExitCode: 0, Out: fmt.Sprintf(`{"created_at":"%v","name":"%v"}`, "0001-01-01T00:00:00Z", bucket)})
```

Comment:
it is weird that gofakes3 returning zero `time.Time{}`, we can investigate it lates why it is empty until then;

if you want to assert time, you can utilize;
https://github.com/peak/s5cmd/blob/8355901a7be5f970a183e7d4d16d1e3b0f2ee840/e2e/sync_test.go#L1009-L1009

or ignore it with prefix matching;
https://github.com/peak/s5cmd/blob/8355901a7be5f970a183e7d4d16d1e3b0f2ee840/e2e/ls_test.go#L110-L119

Instead of building `icmd.Expected`, just assert exit code with `result.Assert` and utilize our output test helpers, you can take a look at `ls` tests;
https://github.com/peak/s5cmd/blob/master/e2e/ls_test.go#L95-L97

Reply:
aws s3api only returns region of bucket in my local too.

```
{
    "BucketRegion": "us-east-1"
}
```
![image](https://github.com/peak/s5cmd/assets/91782773/58a21718-d20f-4af4-b0ed-95f384bb4acf)

Reply:
oh my bad, i've totally missed it, this is on us;
https://github.com/peak/s5cmd/blob/8355901a7be5f970a183e7d4d16d1e3b0f2ee840/storage/storage.go#L200-L203

we have `CreationDate` defined in `Bucket` type and this field is not set in `HeadBucket`;

https://github.com/4o4x/s5cmd/blob/c56f6c0f8b6a1fddc8148d2854cd14bc40ce48b6/storage/s3.go#L1177-L1179

---

first two things comes to my mind is to either;
* make `HeadBucket` return just a string e.g; bucket name.
* ignore `CreationDate` on output if it is empty; either making field pointer with `omitempty` or custom `MarshalJSON` , not sure.